core/new-gui/src/app/dashboard/admin/component/gmail/gmail.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, OnInit } from "@angular/core";
+import { GmailService } from "../../service/gmail.service";
+import { FormBuilder, FormGroup, Validators } from "@angular/forms";
+import { UntilDestroy, untilDestroyed } from "@ngneat/until-destroy";
+@UntilDestroy()
+@Component({
+  selector: "texera-gmail",
+  templateUrl: "./gmail.component.html",
+  styleUrls: ["./gmail.component.scss"],
+})
+export class GmailComponent implements OnInit {
+  public validateForm!: FormGroup;
+  isVisible = false;
+  public email: String | undefined;
+  constructor(private gmailAuthService: GmailService, private formBuilder: FormBuilder) {}
+
+  sendEmail(): void {
+    this.gmailAuthService.sendEmail(
+      this.validateForm.value.subject,
+      this.validateForm.value.content,
+      this.validateForm.value.email
+    );
+    this.isVisible = true;
+  }
+
+  ngOnInit(): void {
+    this.validateForm = this.formBuilder.group({
+      email: [null, [Validators.email, Validators.required]],
+      subject: [null, [Validators.required]],
+      content: [null, [Validators.required]],
+    });
+    this.gmailAuthService.auth();
+    this.getEmail();
+  }
+
+  getEmail() {
+    this.gmailAuthService
+      .getEmail()
+      .pipe(untilDestroyed(this))
+      .subscribe({
+        next: email => (this.email = email),
+        error: (err: unknown) => (this.email = undefined),
+      });
+  }
+
+  public auth() {
```

Comment:
I see two `auth()` in this PR. Another is in `gmail.service.ts`, and it looks like they have different meanings?

Reply:
They are the same.